File: GXpert/GXpert.Web/Modules/Masters/ProjectSkill/ProjectSkillColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace GXpert.Masters.Columns;

[ColumnsScript("Masters.ProjectSkill")]
[BasedOnRow(typeof(ProjectSkillRow), CheckNames = true)]
public class ProjectSkillColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string SkillCategoryName { get; set; }
    public string SkillName { get; set; }
    public string ProjectTitle { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Masters/ProjectSkillColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ProjectSkillRow } from "./ProjectSkillRow";

export interface ProjectSkillColumns {
    Id: Column<ProjectSkillRow>;
    SkillCategoryName: Column<ProjectSkillRow>;
    SkillName: Column<ProjectSkillRow>;
    ProjectTitle: Column<ProjectSkillRow>;
}

export class ProjectSkillColumns extends ColumnsBase<ProjectSkillRow> {
    static readonly columnsKey = 'Masters.ProjectSkill';
    static readonly Fields = fieldsProxy<ProjectSkillColumns>();
}